### DISCUSSION TITLE
请教缓存的使用

### DISCUSSION THREAD

**Discussion by biggbuddy** [kkkgo/PaoPao-Pref#8] [提问和帮助] [answered]
为了用pref生成dns缓存， 单独搭了一套dns和pref。
在生成缓存的时候， 为了得到既有ipv4也有ipv6的最全记录， 在这套环境中把ipv6配置成raw。
缓存生成完成后， 把这个redis_dns_v2.rdb复制到网络中正在使用的dns容器内， 这时如果不考虑域名指向IP变动的情况， 无论后面我把ipv6配置成no,yes,only6,yes_only6,raw 中的任何一个，请问都是可以直接从缓存中命中的吧？

**Comment by kkkgo** [comment]
可以从缓存中命中但结果取决于你的最终选项，比如你设置为no，那绝对不会提供任何AAAA记录。

**Reply by biggbuddy** [reply]
了解。
再问多一嘴，如果一个域名有v4和v6的地址， 并且都保存在了redis缓存文件中， 后面把缓存文件放到了设置为ipv6=no的容器中使用。然后出现了域名指向IP发生了变动的情况， paopaodns应该有机制去更新这个变动吧？这时是不是会因为当前配置是no, 就把这个域名中的ipv6地址删除， 只更新为最新的ipv4地址？

**Reply by kkkgo** [reply, marked as answer]
不是你理解的那样，A记录和AAAA记录没有什么联系，他们是不同的请求，不存在一个请求同时请求A记录和AAAA记录。配置为no仅仅是在你请求AAAA记录的时候执行拒绝。

**Reply by biggbuddy** [reply]
我的理解，比如缓存 pornhub.com 的 v4 和 v6的地址时， 可能是这样的   key: pornhub.com, value:[v4:xxxx, v6:xxx], 而当P站域名对应的IP发生变动时，paopao dns会认为需要查询最新的dns信息, 所以会根据当前配置的ipv6=no去保存 v4的地址， 从而变成 key: pornhub.com, value:[v4:xxxx], 所以 P 站后面无论paopaodns的ipv6再怎么修改它的配置也只有v4的地址了，直到等待下一次的域名更新或类似主动更新缓存的机构， 不是这样的吗？

**Reply by kkkgo** [reply]
都说了没有联系，他们是两个不同的记录。

**Reply by kkkgo** [reply]
你在权威DNS托管的域名的结构是什么样的，实际上就是什么样的。

**Reply by biggbuddy** [reply]
了解了， 感谢
